### PR TITLE
Use system-deps

### PR DESCRIPTION
**Pull request by gdesmott**
Use system-deps instead of our custom deps handing code.

[Tested with gstreamer-sys](https://gitlab.freedesktop.org/gdesmott/gstreamer-rs-sys/pipelines/165046).

cc @sdroege

### CODE REVIEW

**Comment by sdroege** [comment]
Thanks, looks good to me. We should get this in for the release :)

@epashkin @GuillaumeGomez

**Comment by GuillaumeGomez** [comment]
I don't see where it is used in `gstreamer-sys`. It isn't in the `cargo.toml` file?

**Comment by sdroege** [comment]
> I don't see where it is used in `gstreamer-sys`. It isn't in the `cargo.toml` file?

@gdesmott didn't send an MR for it yet because it's not merged into gir yet

**Comment by GuillaumeGomez** [comment]
I have troubles with merging something this huge just before a new release.

**Comment by sdroege** [comment]
Functionally it's the same we had before by default, but it a) moves the logic into another crate so we don't have to copy it everywhere and b) allows overriding the behaviour in various ways via environment variables if the default is not good

File: src/codegen/sys/cargo_toml.rs [hunk 1]
```diff
@@ -82,7 +82,7 @@ fn fill_in(root: &mut Table, env: &Env) {
 
     {
         let build_deps = upsert_table(root, "build-dependencies");
-        set_string(build_deps, "pkg-config", "0.3.7");
+        set_string(build_deps, "system-deps", "1.3");
```
Comment (EPashkin) [inline src/codegen/sys/cargo_toml.rs:85]: Maybe add `unset(build_deps,  "pkg-config");` as it no more needed

**Comment by EPashkin** [comment]
@gdesmott Thans, it works fine on my Windows/MSys64.

File: src/codegen/sys/cargo_toml.rs (hunk 1 above)
Comment (gdesmott) [reply src/codegen/sys/cargo_toml.rs:85]: Good point, I added that.

**Comment by EPashkin** [comment]
@gdesmott Thanks again

File: src/codegen/sys/cargo_toml.rs [hunk 2]
```diff
@@ -82,7 +82,8 @@ fn fill_in(root: &mut Table, env: &Env) {
...
+        unset(build_deps, "pkg-config");
```
Comment (sdroege) [inline src/codegen/sys/cargo_toml.rs:86]: This breaks `gdk-sys`, which uses `pkg-config` in some custom build.rs code.

See CI failure that will happen here soon: https://github.com/gtk-rs/sys/pull/174 (`Cargo.toml` will have changes after running gir: the removed `pkg-config` dependency)

Comment (EPashkin) [reply src/codegen/sys/cargo_toml.rs:86]: Добавил https://github.com/gtk-rs/gir/pull/941,

Comment (sdroege) [reply src/codegen/sys/cargo_toml.rs:86]: Спасибо :)

Comment (GuillaumeGomez) [reply src/codegen/sys/cargo_toml.rs:86]: Русское время!

**Pull request merged**